tests/codegen/thread-local.rs

```diff
@@ -14,8 +14,8 @@ use std::cell::Cell;
 
 thread_local!(static A: Cell<u32> = const { Cell::new(1) });
 
-// CHECK: [[TLS_AUX:@.+]] = external thread_local local_unnamed_addr global i64
-// CHECK: [[TLS:@.+]] = internal thread_local unnamed_addr global
+// CHECK: [[TLS_AUX:@.+]] = external thread_local global i64
+// CHECK: [[TLS:@.+]] = internal thread_local global
```

Comment:
This change to thread_local codegen seems unintended.

Reply:
Agreed, unfortunately I really don't understand what's causing this: I've removed the changes where I was changing how `unnamed_addr` was applied, yet this diff still appears...

Reply:
Looks like it wasn't added by rustc, but by LLVM's GlobalOpt pass: https://godbolt.org/z/zh6cPsra4

It looks like GlobalOpt is inibited here because a function that returns the thread local's address is kept alive only by a `weak_odr` global. On my machine, using `linkonce_odr` fixes this as well.

Reply:
Yep, just noticed that when running the tests with `linkonce_odr` - thanks again for that suggestion!